yb-voyager/cmd/common.go

```diff
@@ -970,6 +970,39 @@ func storeTableListInMSR(tableList []sqlname.NameTuple) error {
 	return nil
 }
 
+var appChanges = []string{
+	INHERITANCE_ISSUE_REASON,
+	CONVERSION_ISSUE_REASON,
+	DEFERRABLE_CONSTRAINT_ISSUE,
+	fmt.Sprintf("%s - xml", UNSUPPORTED_DATATYPE),
+	fmt.Sprintf("%s - xid", UNSUPPORTED_DATATYPE),
+	UNSUPPORTED_EXTENSION_ISSUE, // will confirm this
+}
+
+// Migration complexity calculation from the conversion issues
+func getMigrationComplexity(sourceDBType string) string {
+	if sourceDBType != POSTGRESQL {
+		return "NOT AVAILABLE"
+	}
+	log.Infof("Calculating migration complexity..")
+	appChangesCount := 0
+	for _, issue := range schemaAnalysisReport.Issues {
+		for _, appChange := range appChanges {
+			if strings.Contains(issue.Reason, appChange) {
+				appChangesCount++
+			}
+		}
+	}
+	schemaChangesCount := len(schemaAnalysisReport.Issues) - appChangesCount
+	if (appChangesCount >= 1 && appChangesCount < 5) || (schemaChangesCount >= 20) {
+		return MEDIUM
+	} else if appChangesCount > 5 {
+		return HIGH
+	}
+	//LOW in case appChanges == 0 or schemaChanges [0-20)
+	return LOW
```

Comment:
lets define constants here since chances of these values getting updated is higher:

appChangesThresholdHigh = 5
appChangesThresholdMedium = 1
schemaChangesThresholdMedium = 20
....

Reply:
Yes will define that. 
Also, I need to support ENV in changing that for the short term for testing.